src/libraries/System.Net.Sockets/tests/FunctionalTests/Connected.cs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+
+namespace System.Net.Sockets.Tests
+{
+    public class Connected
+    {
+        [Fact]
+        public void NonBlockingFailedConnect_ConnectedReturnsFalse()
+        {
+            using Socket socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+
+            socket.Blocking = false;
+
+            // Connect to port 1 where we expect no server to be listening.
+            SocketException se = Assert.ThrowsAny<SocketException>(() => socket.Connect(IPAddress.Loopback, 1));
```

Comment:
Personally I'm fine with this, but I remember someone complaining about magic port numbers in a PR review before, and suggesting to create & bind a temporary socket instead.

Reply:
In this case, we want a port that is ensured to not be listening. We can't leverage `TCP_WAIT` because we unconditionally set `SO_REUSEADDR` on `Bind` for Windows compatibility.

https://github.com/dotnet/runtime/blob/8455a5edd70a42c06eb2320c1c5656fd13ba8fe2/src/libraries/Native/Unix/System.Native/pal_networking.c#L1553-L1559

Port number `1` is a reserved port for a service that is no longer used. So it's a good pick.